CMake/scipp-util.cmake

```diff
@@ -21,10 +21,22 @@ function(scipp_function template category function_name)
         PARENT_SCOPE
     )
   endmacro()
+
+  set(options SKIP_VARIABLE NO_OUT)
+  cmake_parse_arguments(PARSE_ARGV 3 SCIPP_FUNCTION "${options}" "" "" )
+
   message("Generating files for ${function_name}")
   set(NAME ${function_name})
   set(ELEMENT_INCLUDE ${category})
-  configure_in_module("variable" ${function_name})
+  if(SCIPP_FUNCTION_NO_OUT)
+    set(GENERATE_OUT "false")
+  else()
+    set(GENERATE_OUT "true")
+  endif()
+
+  if(NOT SCIPP_FUNCTION_SKIP_VARIABLE)
+    configure_in_module("variable" ${function_name})
+  endif()
   configure_in_module("dataset" ${function_name})
   set(src ${NAME}.cpp)
   configure_file(templates/python_${template}.cpp.in python/${src})
```

Comment:
This is still generating bindings for variables even if `SKIP_VARIABLE` is set. Could there be situations where a `Dataset` function and binding are trivial but both `Variable` function _and_ binding need special attention?

Reply:
Generating Python bindings for variables is definitely deliberate here.

The only special attention I can think of right now is that we generally support `out` args for variable functions, but not for data array functions. This is handled in the template using an `if constexpr`.

Of course we still have (and probably will always have) numerous cases that are not covered by the templates (neither for variable, for data arrays, nor for bindings), so we do them by hand.